server/portal/apps/workspace/api/views.py

```diff
@@ -151,6 +141,54 @@ def get(self, request, *args, **kwargs):
             encoder=BaseTapisResultSerializer
         )
 
+    def select(self, client, request):
+        job_uuid = request.GET.get('job_uuid')
+        data = client.jobs.getJob(jobUuid=job_uuid)
+
+        return data
+
+    def listing(self, client, request):
+        limit = int(request.GET.get('limit', 10))
+        offset = int(request.GET.get('offset', 0))
+        portal_name = settings.PORTAL_NAMESPACE
+
+        data = client.jobs.getJobSearchList(
+            limit=limit,
+            startAfter=offset,
+            orderBy='lastUpdated(desc),name(asc)',
+            _tapis_query_parameters={'tags.contains': f'portalName: {portal_name}'},
+            select='allAttributes'
+        )
+
+        return data
+
+    def search(self, client, request):
+
+        query_string = request.GET.get('query_string')
+
+        limit = int(request.GET.get('limit', 10))
+        offset = int(request.GET.get('offset', 0))
+        portal_name = settings.PORTAL_NAMESPACE
+
+        sql_queries = [
+            f"(tags IN ('portalName: {portal_name}')) AND",
+            f"(name like '%{query_string}%') OR",
+            f"(archiveSystemDir like '%{query_string}%') OR",
+            f"(appId like '%{query_string}%') OR",
+            f"(archiveSystemId like '%{query_string}%')",
+        ]
+
+        data = client.jobs.getJobSearchListByPostSqlStr(
+            limit=limit,
+            startAfter=offset,
+            orderBy='lastUpdated(desc),name(asc)',
+            request_body={
+                "search": sql_queries
+            }
```

Comment:
There are some extra fields we need from the jobs objects in order to calculate the output directory. This should allow the job search results to still show the output directory
```suggestion
            },
            select="allAttributes"
```